# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/RenderRootStateManager.h
# Commit: c09a51622e98
# Full Hash: c09a51622e988e81077c75e3aaa5a49df66284b3
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2019-03-22 16:44:08
# Regressor Bug: 1441308
# File Overlap Count: 2
# Description:
#   Bug 1441308 - Core renderroot splitting changes r=kats,sotaro
#   
#   This is a large patch that contains all of the core changes for
#   renderroot splitting.
#   
# ==============================================================================

diff -r 737807563dd5 -r c09a51622e98 gfx/layers/wr/RenderRootStateManager.h
--- a/gfx/layers/wr/RenderRootStateManager.h	Thu Mar 21 23:14:26 2019 +0000
+++ b/gfx/layers/wr/RenderRootStateManager.h	Fri Mar 22 03:15:14 2019 +0000
@@ -10,6 +10,7 @@
 #include "mozilla/webrender/WebRenderAPI.h"
 
 #include "mozilla/layers/IpcResourceUpdateQueue.h"
+#include "mozilla/layers/SharedSurfacesChild.h"
 #include "mozilla/layers/WebRenderCommandBuilder.h"
 
 namespace mozilla {
@@ -24,10 +25,14 @@
   void AddRef();
   void Release();
 
-  explicit RenderRootStateManager(WebRenderLayerManager* aLayerManager);
+  RenderRootStateManager()
+      : mLayerManager(nullptr),
+        mRenderRoot(wr::RenderRoot::Default),
+        mDestroyed(false) {}
 
   void Destroy();
   bool IsDestroyed() { return mDestroyed; }
+  wr::RenderRoot GetRenderRoot() { return mRenderRoot; }
   wr::IpcResourceUpdateQueue& AsyncResourceUpdates();
   WebRenderBridgeChild* WrBridge() const;
   WebRenderCommandBuilder& CommandBuilder();
@@ -74,7 +79,6 @@
   void FlushAsyncResourceUpdates();
 
  private:
-  ~RenderRootStateManager();
   WebRenderLayerManager* mLayerManager;
   Maybe<wr::IpcResourceUpdateQueue> mAsyncResourceUpdates;
   nsTArray<wr::ImageKey> mImageKeysToDelete;
@@ -89,6 +93,7 @@
   // the compositor to discard information for.
   nsTArray<uint64_t> mDiscardedCompositorAnimationsIds;
 
+  wr::RenderRoot mRenderRoot;
   bool mDestroyed;
 
   friend class WebRenderLayerManager;